src/modules/commander/Arming/PreFlightCheck/PreFlightCheck.cpp

```diff
@@ -691,286 +1226,37 @@ static bool failureDetectorCheck(orb_advert_t *mavlink_log_pub, const vehicle_st
-
-				if ((prime_id > 0) && (device_id == prime_id)) {
-					prime_found = true;
-				}
-
-			} else {
-				if (required) {
-					failed = true;
-					mag_fail_reported = true;
-				}
-			}
-		}
-
-		if (sys_has_mag == 1) {
-			/* check if the primary device is present */
-			if (!prime_found) {
-				if (reportFailures && !failed) {
-					mavlink_log_critical(mavlink_log_pub, "Primary compass not found");
-				}
-
-				set_health_flags(subsystem_info_s::SUBSYSTEM_TYPE_MAG, false, true, false, status);
-				failed = true;
-			}
-
-			/* mag consistency checks (need to be performed after the individual checks) */
-			if (!magConsistencyCheck(mavlink_log_pub, status, (reportFailures && !failed))) {
-				failed = true;
-			}
-		}
-	}
-
-	/* ---- ACCEL ---- */
-	if (checkSensors) {
-		bool prime_found = false;
-		int32_t prime_id = -1;
-		param_get(param_find("CAL_ACC_PRIME"), &prime_id);
-
-		bool accel_fail_reported = false;
-
-		/* check all sensors individually, but fail only for mandatory ones */
-		for (unsigned i = 0; i < max_optional_accel_count; i++) {
-			const bool required = (i < max_mandatory_accel_count);
-			const bool report_fail = (reportFailures && !failed && !accel_fail_reported);
-
-			int32_t device_id = -1;
-
-			if (accelerometerCheck(mavlink_log_pub, status, i, !required, checkDynamic, device_id, report_fail)) {
-
-				if ((prime_id > 0) && (device_id == prime_id)) {
-					prime_found = true;
-				}
-
-			} else {
-				if (required) {
-					failed = true;
-					accel_fail_reported = true;
-				}
-			}
-		}
-
-		/* check if the primary device is present */
-		if (!prime_found) {
-			if (reportFailures && !failed) {
-				mavlink_log_critical(mavlink_log_pub, "Primary accelerometer not found");
-			}
-
-			set_health_flags(subsystem_info_s::SUBSYSTEM_TYPE_ACC, false, true, false, status);
-			failed = true;
-		}
-	}
-
-	/* ---- GYRO ---- */
-	if (checkSensors) {
-		bool prime_found = false;
-		int32_t prime_id = -1;
-		param_get(param_find("CAL_GYRO_PRIME"), &prime_id);
-
-		bool gyro_fail_reported = false;
-
-		/* check all sensors individually, but fail only for mandatory ones */
-		for (unsigned i = 0; i < max_optional_gyro_count; i++) {
-			const bool required = (i < max_mandatory_gyro_count);
-			const bool report_fail = (reportFailures && !failed && !gyro_fail_reported);
-
-			int32_t device_id = -1;
-
-			if (gyroCheck(mavlink_log_pub, status, i, !required, device_id, report_fail)) {
-
-				if ((prime_id > 0) && (device_id == prime_id)) {
-					prime_found = true;
-				}
-
-			} else {
-				if (required) {
-					failed = true;
-					gyro_fail_reported = true;
-				}
-			}
-		}
-
-		/* check if the primary device is present */
-		if (!prime_found) {
-			if (reportFailures && !failed) {
-				mavlink_log_critical(mavlink_log_pub, "Primary gyro not found");
-			}
-
-			set_health_flags(subsystem_info_s::SUBSYSTEM_TYPE_GYRO, false, true, false, status);
-			failed = true;
-		}
-	}
-
-	/* ---- BARO ---- */
-	if (checkSensors) {
```

Comment:
Perhaps not what you want to tackle here, but I'm wondering if all this sensors code should be owned by the sensors module. Then a higher level sensors status would be published for commander to act on.